services/inputhost/pubconnection.go

```diff
@@ -504,17 +510,61 @@ func (conn *pubConnection) flushCmdToClient(unflushedWrites int) (err error) {
 	return
 }
 
-func (conn *pubConnection) writeCmdToClient(cmd *cherami.InputHostCommand) (err error) {
+func (conn *pubConnection) writeCmdToClient(cmd *cherami.InputHostCommand, drainWG *sync.WaitGroup) (err error) {
 	if err = conn.stream.Write(cmd); err != nil {
```

Comment:
mark the WG as done even when the write fails, that way, the caller doesn't have to await until timeout